### ISSUE TITLE
[BUG]Not able to apply hires fix within webui with comfyui extension turned on.

### ISSUE THREAD

**Issue by anitman** [ModelSurge/sd-webui-comfyui#140]
It seems that this extension can't work with hires fix. After disabling the extension, the img can be generated with no problem.

>    
      File "E:\git\stable-diffusion-webui\modules\call_queue.py", line 58, in f
        res = list(func(*args, **kwargs))
      File "E:\git\stable-diffusion-webui\modules\call_queue.py", line 37, in f
        res = func(*args, **kwargs)
      File "E:\git\stable-diffusion-webui\modules\txt2img.py", line 62, in txt2img
        processed = processing.process_images(p)
      File "E:\git\stable-diffusion-webui\modules\processing.py", line 677, in process_images
        res = process_images_inner(p)
      File "E:\git\stable-diffusion-webui\modules\processing.py", line 794, in process_images_inner
        samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
      File "E:\git\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\webui\workflow_patcher.py", line 85, in p_sample_patch
        x = original_function(*args, **kwargs)
      File "E:\git\stable-diffusion-webui\modules\processing.py", line 1142, in sample
        samples = self.sampler.sample_img2img(self, samples, noise, self.hr_c, self.hr_uc, steps=self.hr_second_pass_steps or self.steps, image_conditioning=image_conditioning)
      File "E:\git\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ipc\__init__.py", line 41, in wrapper
        return function(*args, **kwargs)
      File "E:\git\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\webui\workflow_patcher.py", line 60, in sample_img2img_hijack
        batch_input=type_conversion.webui_latent_to_comfyui(x).to(device='cpu'),
    AttributeError: 'dict' object has no attribute 'to'

I also noted that batch_input's device equals 'cpu' not 'cuda'.

**Comment by ljleb** [comment]
The code is applying the `to('cpu')` on the dict instead of the tensor inside, slight oversight on my part.

**Comment by John-WL** [comment]
We are applying `to('cpu')` in order to be able to share the tensors between processes, so the switch to cpu is expected.

**Comment by anitman** [comment]
In my understanding, the basic image is generated in the GPU's memory. However, during the "hires fix" latent process, it should also access the GPU memory. But since the plugin looks for it in the system memory where the image cannot be found, this results in an error. Is there an issue with my understanding?

**Comment by ljleb** [comment]
Still does not work, after testing. I changed the conversion to take batches as input.

> Is there an issue with my understanding?

The webui takes whatever the extensions return. while the extension code is running, we can do whatever we want with the tensors, delete them and put new ones, there isn't any issue with that. It's just the conversion code that's a bit funky because I went in diagonal over this part of the code when refactoring the workflow types external_code API.
